### PR TITLE
Add support for AMD modifiers

### PR DESCRIPTION
**Pull request by emersion**
~~Depends on https://github.com/ascent12/drm_info/pull/60~~

It's not possible in practice to explicitly enumerate all modifiers, because they are very complicated. Instead, rely on drm_fourcc.h macros and manually format a string.

Closes: #62

cc @bnieuwenhuizen

### CODE REVIEW

File: modifiers.c [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+			printf(", DCC_PIPE_ALIGN");
+		}
+		if (AMD_FMT_MOD_GET(DCC_INDEPENDENT_64B, mod)) {
+			printf(", DCC_INDEPENDENT_64B");
+		}
+		if (AMD_FMT_MOD_GET(DCC_INDEPENDENT_128B, mod)) {
+			printf(", DCC_INDEPENDENT_128B");
+		}
+		if (AMD_FMT_MOD_GET(DCC_MAX_COMPRESSED_BLOCK, mod)) {
+			printf(", DCC_MAX_COMPRESSED_BLOCK");
```
Comment (BNieuwenhuizen) [inline modifiers.c:77]: This is not a boolean (0 = 64 bytes, 1 = 128 bytes, 2 = 256 bytes).

Of course I just noticed that the mask in the headers is 0x1 and the next bit contains another field ...

File: modifiers.c [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+	uint64_t tile_version = AMD_FMT_MOD_GET(TILE_VERSION, mod);
+	uint64_t tile = AMD_FMT_MOD_GET(TILE, mod);
+	uint64_t dcc = AMD_FMT_MOD_GET(DCC, mod);
+	uint64_t dcc_retile = AMD_FMT_MOD_GET(DCC_RETILE, mod);
+
+	printf("TILE_VERSION = %s", amd_tile_version_str(tile_version));
+	printf(", TILE = %s", amd_tile_str(tile, tile_version));
+
+	if (dcc) {
+		printf(", DCC");
```
Comment (BNieuwenhuizen) [inline modifiers.c:66]: Maybe print "DCC_RETILE" if it is set? I think you don't print it anywhere?

Comment (emersion) [reply modifiers.c:76]: Oh, right, this is missing. Fixed!

File: modifiers.c (hunk 1 above)
Comment (emersion) [reply modifiers.c]: Updated to use the enum, even if drm-next still has the broken version of this.

**Comment by BNieuwenhuizen** [comment]
LGTM

**Comment by BNieuwenhuizen** [comment]
btw for landing this it sounds like we'll also want to sync the headers to libdrm after the fix is in (mesa now uses its own copy for the headers so I hadn't put any effort into that yet)

**Review by ascent12 — APPROVED** [review]
LGTM.
Feel free to ping me or merge it yourself when the required things are in an upstream libdrm release.

**Comment by emersion** [comment]
In released libdrm 2.4.104 with the updated kernel headers.

**Pull request merged**